arch/x86/core/Kconfig.ia32

```diff
@@ -35,6 +35,7 @@ config SET_GDT
 config GDT_DYNAMIC
 	bool "Store GDT in RAM so that it can be modified"
 	depends on SET_GDT
+	default y if THREAD_LOCAL_STORAGE
```

Comment:
does this need to be stronger? I think we *must* have a RAM-based GDT in order for this to work...is there a way we can make CONFIG_THREAD_LOCAL_STORAGE enforce that GDT_DYNAMIC and SET_GDT are turned on?

I had a similar problem with userspace, I ended up creating a hidden X86_USERSPACE config which turned on the stuff I needed:

```
config X86_USERSPACE
	bool
	default y if USERSPACE
	select THREAD_STACK_INFO
	select SET_GDT
	select GDT_DYNAMIC
	select X86_ENABLE_TSS
	help
	  This option enables APIs to drop a thread's privileges down to ring 3,
	  supporting user-level threads that are protected from each other and
	  from crashing the kernel.
```

Reply:
I was trying not to select `GDT_DYNAMIC` inside `kernel/kconfig`. I will try the hidden kconfig.